android/app/src/main/java/org/fossasia/openevent/fragments/AboutFragment.java

```diff
@@ -254,6 +256,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 displaySpeakersCallInformation();
                 break;
             case R.id.action_download_latest_data:
+                SharedPreferencesUtil.putBoolean(ConstantStrings.IS_DOWNLOAD_DONE, false);
```

Comment:
This is not the correct way. Download may be done. Saving the wrong information is ... wrong. Simply show the dialog even if the data is downloaded, i.e., don't check if it is downloaded. 

Second this you would have to do is that currently, if user denies to download latest data, it is loaded from JSON. But if the data is already downloaded, it should not be loaded from JSON, so when handling negative response, add the check if the data is downloaded. If it is not, then load from JSON, otherwise do nothing

Reply:
@iamareebjamal Yeah true! Should I create a separate method that will take into account everything you mentioned then?

Reply:
Yes